src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/TraceEventExtensions.cs

```diff
@@ -2,11 +2,17 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using Microsoft.Diagnostics.Tracing;
+using System;
 
 namespace Microsoft.Diagnostics.Monitoring.WebApi
 {
     internal static class TraceEventExtensions
     {
+        public static bool GetBoolPayload(this TraceEvent traceEvent, int index)
+        {
+            return Convert.ToBoolean(GetPayload<uint>(traceEvent, index));
+        }
+
         public static T GetPayload<T>(this TraceEvent traceEvent, int index)
         {
             return (T)traceEvent.PayloadValue(index);
```

Comment:
note: I plan on updating this with a `DEBUG` check to prevent us accidentally calling it with `<bool>`. This will happen in a separate PR since we have an instance of using `<bool>` here elsewhere in dotnet-monitor that needs to get fixed.

Reply:
I don't suppose there is a way to special case this such that if it is called with `bool` that we just do the specialization instead of the generic fallback?

Reply:
I initially tried to do that, but wasn't able to figure a way of convincing msbuild to do the necessary type conversions short of something like this which I'm not a massive fan of:
```csharp
        public static T GetPayload<T>(this TraceEvent traceEvent, int index)
        {
            if (typeof(T) == typeof(bool))
            {
                return (T)(object)Convert.ToBoolean(GetPayload<uint>(traceEvent, index));
            }
            ...
```

With regards to this PR though, I can't change the default behavior of `GetPayload<bool>` since we have other code incorrectly using it today that needs to get fixed first. If there's a way of nicely special casing `GetPayload<bool>` we can do it in follow-up PR that fixes the existing code that uses it.